okhttp/src/main/java/io/grpc/okhttp/OkHttpProtocolNegotiator.java

```diff
@@ -133,6 +140,59 @@ public String getSelectedProtocol(SSLSocket socket) {
     private static final OptionalMethod<Socket> SET_NPN_PROTOCOLS =
         new OptionalMethod<>(null, "setNpnProtocols", byte[].class);
 
+    // Available on Android 10.0+.
+    // SSLParameters#setApplicationProtocols(String[])
+    private static final OptionalMethod<SSLParameters> SET_APPLICATION_PROTOCOLS =
```

Comment:
It'd be better to use the normal reflection `Method` here. These `OptionalMethod`s were for the duck-typing and do reflection every call. It'd be much better to only do the reflection once.

Although we may also want to defer that reflection until first use...

Reply:
As discussed separately, since this is in the negotiator which is loaded late it is actually okay to avoid deferring until first use. I keep thinking we are in Platform which is loaded during Channel creation.

Reply:
Changed to use normal `Method`, as those are public APIs.  One downside is we are no longer able to have unit tests that verifies the invocation.